crates/iota-grpc-api/src/event_service.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use iota_json_rpc_types::{EventFilter, Filter, IotaEvent};
+use iota_types::{
+    base_types::{IotaAddress, ObjectID},
+    digests::TransactionDigest,
+};
+use move_core_types::{identifier::Identifier, language_storage::StructTag};
+use serde_json;
+use tokio_util::sync::CancellationToken;
+use tonic::{Request, Response, Status};
+use tracing::debug;
+
+use crate::{
+    bcs_event::try_to_bcs_bytes,
+    events::{Event, EventId, EventStreamRequest, event_service_server::EventService},
+    types::GrpcEventBroadcaster,
+};
+
+pub struct EventGrpcService {
+    pub event_broadcaster: GrpcEventBroadcaster,
```

Comment:
Should it be like this:

```suggestion
pub struct EventGrpcService<T: EventBroadcaster> {
    pub event_broadcaster: T,
```

so that `EventBroadcaster` trait is justified?

Reply:
Thanks for the comment. We removed this one and only have subscriber now in [f06a81d](https://github.com/iotaledger/iota/pull/7746/commits/f06a81d38dd7d978820da6fa3957d8f1c019f214).